prompt2model/dataset_transformer/prompt_based.py

```diff
@@ -0,0 +1,123 @@
+"""A simple dataset transformer that uses a plan prompt and transform prompt."""
+
+from __future__ import annotations
+
+import asyncio
+
+import datasets
+
+from prompt2model.dataset_transformer.base import DatasetTransformer
+from prompt2model.dataset_transformer.prompt_template import (
+    construct_prompt_for_plan,
+    construct_prompt_for_transform_data,
+)
+from prompt2model.prompt_parser import PromptSpec
+from prompt2model.utils import (
+    API_ERRORS,
+    api_tools,
+    get_formatted_logger,
+    handle_api_error,
+)
+from prompt2model.utils.parse_json_responses import (
+    extract_response,
+    make_request_from_prompt,
+)
+
+logger = get_formatted_logger("DatasetTransformer")
+
+
+class PromptBasedDatasetTransformer(DatasetTransformer):
+    """A class for transforming given dataset to required format using a plan and a transform prompt."""  # noqa E501
```

Comment:
Avoid noqas unless really necessary; linting exists for a reason (because it helps readability). They're mostly not necessary in this file.

```suggestion
    """Transform data based on a transform prompt."""
```